src/engine/ui/widgets/Slider.cxx

```diff
@@ -82,7 +92,39 @@ void Slider::onMouseMove(const SDL_Event &event)
       sliderButton.x = event.motion.x - sliderButton.w / 2;
       curVal = getValue(event.motion.x);
     }
+
+    //move all the elements being slid by this slider
+    for(auto elem : m_uiElements)
+    {
+      SDL_Rect r = elem->getUiBaseElementRect();
+      elem->setPosition(r.x + curVal - m_maxVal/2, r.y);
+    }
+  }
+}
+
+bool Slider::isMouseOver(int x, int y) { return overSliderButton(x, y) || overSliderLine(x, y); }
+
+void Slider::addUiElement(UIElement *elem){
+  //add the new element to m_uiElements
```

Comment:
I assume this is for moving other UI elemens (like the buttons in the menu) with the slider.
Functionality like this should NEVER be part of a widget. the widgets is solely responsible for drawing and stuff. Callback functions like that must be added with registerCallback.

Reply:
Yeah, i was assuming that the sliders purpose was to move other elements. This function registers the elements that would be moved by this slider. This couldn't be done on callback unless there was a callback for onLoad and even then, the elements would have to be tracked by the UIManager, it'd be ugly. This functionality should exist in the Frame widget but that'll be a big change.